### PR TITLE
Fixes bug when failing predicate was not stoping reducer to be executed

### PR DESCRIPTION
**Pull request by adiki**
When playing with example I discovered that predicate is not applied when tapping plus button multiple times.
I did fix that issue and implement test which is covering that case.

### CODE REVIEW

**Review by sergdort — APPROVED** [review]
👍

File: ReactiveFeedbackTests/SystemTests.swift [hunk 1]
```diff
@@ -216,4 +216,48 @@ class SystemTests: XCTestCase {
...
+            guard numberOfTimes > 0 else {
+                token.dispose()
+                return
+            }
+            DispatchQueue.main.async {
+                incrementObserver.send(value: ())
+                increment(numberOfTimes: numberOfTimes - 1)
+            }
+        }
+        increment(numberOfTimes: 3)
```
Comment (RuiAAPeres) [inline ReactiveFeedbackTests/SystemTests.swift]: Shouldn't you increment this more than 3 times, so we can be sure that the Feedback isn't called more than 3 due to the `predicate`?

Comment (adiki) [reply ReactiveFeedbackTests/SystemTests.swift]: Incrementing it 3 times results in incrementing it one time above the predicate. It works and  prevents state to be increased above 2. I will make it higher to have stronger test.

**Review by RuiAAPeres — APPROVED** [review]

File: ReactiveFeedback/Feedback.swift [hunk 2]
```diff
@@ -94,8 +94,10 @@ public struct Feedback<State, Event> {
         predicate: @escaping (State) -> Bool,
         effects: @escaping (State) -> Effect
     ) where Effect.Value == Event, Effect.Error == NoError {
-        self.init(deriving: { $0.filter(predicate) },
-                  effects: effects)
+        self.init(deriving: { $0 },
+                  effects: { state -> SignalProducer<Event, NoError> in
+                      predicate(state) ? effects(state).producer : .empty
```
Comment (inamiy) [inline ReactiveFeedback/Feedback.swift:99]: I guess this new implementation is just the same as `init(lensing: { $0 } effects: effects)`.

By modifying to current code f4905c7 , we lost the original filtering logic that once **didn't cancel previous effect when predicate failed**.

**Pull request merged**